src/linker/Linker/Driver.cs

```diff
@@ -343,6 +343,17 @@ protected int SetupContext (ILogger customLogger = null)
 
 						continue;
 
+					case "--set-feature":
+						if (!GetStringParam (token, feature => {
+							if (!GetStringParam (token, value => {
```

Comment:
The idea as described [here](https://github.com/dotnet/designs/pull/99/files#diff-c53a2566ab58ccf823f487a837d80088R95) is that any values can be passed in - but only those that match feature definitions in xml will be applied. If we were to check for supported feature names/values up-front, then we would need to parse all features out of embedded xml files before doing the check, and also maintain a separate list of feature settings as a subset of `RuntimeHostConfigurationOptions` to avoid passing in unsupported names/values. The way I think of it is that all user-facing settings that are known as build time are made available, and we can light up optimizations based on them by defining corresponding features in the embedded xml.

Reply:
There is a pretty strong validation of allowed values in XML which used for comparison at https://github.com/mono/linker/pull/1132/files#diff-65782fe8cce0bb40f3c81724141aa103R56. I think it'd be better to report back to users errors on both sides when it's clear what they passed in won't ever work (right now any non-boolean value).

Reply:
I can understand the desire for it to let you know if you passed some feature settings that will have no effect. The issue is that we are going to flow existing `RuntimeHostConfigurationOptions` which may not have substitution implementations, and may not be boolean, to this input.

I would be fine with logging a low-importance message for inputs that aren't boolean, but it shouldn't show up by default when running from MSBuild because that would result in non-actionable warnings on every linker invocation - and it also wouldn't be able to detect booleans that don't have any defined feature substitutions.

Reply:
> The issue is that we are going to flow existing RuntimeHostConfigurationOptions which may not have substitution implementations, and may not be boolean, to this input.

Should we block them and not send them to illinker at all or the other way should we whitelist RuntimeHostConfigurationOptions which can flow to linker?

/cc @vitek-karas

Reply:
Per our offline discussion I think a good solution would be:
* Filter out non-boolean runtime properties in the ILLinkTask/SDK before passing them into linker as feature values - linker doesn't support them anyway.
* Linker will fail (error) if a non-boolean value is specified for a feature value. The logic behind this is that if someone runs linker directly from command line linker should fail on non-boolean values as it doesn't support them.

Going forward we could also:
* Introduce some type of metadata into the RuntimeHostConfigurationOption item group which would mark certain items as not-features and we would filter those out as well (regardless of their value)
* If / when we allow additional data types it will be a very explicit change (which is a good thing)

Reply:
I made it an error to pass non-bools.